### PR TITLE
Make segment granularity adjustable

### PR DESCRIPTION
**Pull request by eibach**
Segments in x86 real mode have 16 byte granularity(4 address bits).
There are however systems (namely the Vautomation/ARC Turbo 186 cores) that use 256 byte(8 address bits) instead to allow access to 16MiB memory.

Unfortunately fixed granularity is hard coded in lots of places in r2.

Try to fix this up in as many places as possible.

### CODE REVIEW

File: libr/anal/p/anal_x86_udis.c [hunk 1]
```diff
@@ -133,11 +133,11 @@ static int getarg(char *src, struct ud *u, st64 mask, int idx, int regsz) {
 	return 0;
 }
 
-static st64 getval(ud_operand_t *op) {
+static st64 getval(ud_operand_t *op, unsigned int seggranu) {
 	int bits = op->size;
 	switch (op->type) {
 	case UD_OP_PTR:
-		return (op->lval.ptr.seg<<4) + (op->lval.ptr.off & 0xFFFF);
+		return (op->lval.ptr.seg<<seggranu) + (op->lval.ptr.off & 0xFFFF);
```
Comment (XVilka) [inline libr/anal/p/anal_x86_udis.c]: can you use space here `.seg << seggran`

**Comment by XVilka** [comment]
Can you use better name? `seggrn` or `seggran` at least?

**Comment by radare** [comment]
The changs in udis are too intrussive because its adding another argument in many places. Would be good to find a better approach like global variable or set it inside the operand struct. 

I know that udis is no longer maintained. But will be good to not make heavy changes like this to their codebase to simplify merging if someone ever takes it back.

Also, the default disassembler is capstone. Will be good to fill an issue in there if not implemented.

**Comment by eibach** [comment]
I added support for capstone.
Should I simply drop udis support? It is not really important to me. And if intrusive changes are not helpful I don't see much I can do. Global variables are really ugly and changing the operand struct is also quite intrusive. 
Regarding naming I'm quite emotionless. Any votes from pancake?

**Comment by XVilka** [comment]
@eibach I think yes, kill the udis86 support and we should move udis into radare2-extras, since it is unmaintained for quite some time.

**Comment by eibach** [comment]
I dropped udis support.
I changed seggranu to seggrn in all places.

**Comment by radare** [comment]
Awesome thanks!. I will evaluate moving udis to extras , any comments on this?

**Pull request merged**